course02/homework00/joost/vector.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <array>
+#include <stdexcept>
+
+class VectorInt
+{
+	public:
+		VectorInt():
+			mSize(0)
+		{}
+		
+		int size() const
+		{
+			return mSize;
+		}
+
+		int capacity() const
+		{
+			return mData.size();
+		}
+
+		void push_back(int value)
+		{
+			if(mSize == mData.size()){
+				throw std::out_of_range("vector at capacity");
+			}
+			mData[mSize++] = value;
+		}
+
+		int back(){
+			if(mSize == 0){
+				throw std::out_of_range("vector is empty");
+			}
+			return mData.at(--mSize);
```

Comment:
OK now I see that is might not be `const` due to this - it's actually mutable. Well then I would say the API is not great, `back()` should not pop a value - `pop_back()` is taking care of this operation